src/App.tsx

```diff
@@ -225,27 +234,42 @@ export default class App extends React.Component<{}, { expandFolderBrowser: bool
     }
   }
 
-  onOkClick = async (ev: Event) => {
-    const selectedNode = (ev as CustomEvent<TreeNode>).detail;
-    const breadcrumbs = this.folderBrowser.current?.breadcrumbs;
-    const entryId = Number.parseInt(selectedNode.id, 10);
+  onSelectFolder = async () => {
+    if (!this.repoClient) {
+      throw new Error('Repo Client is undefined.');
+    }
+    if (!this.loginComponent) {
+      throw new Error('LfLoginComponent is not found.');
+    }
+    const selectedNode = this.repositoryBrowser.current?.currentFolder as LfRepoTreeNode;
+    let entryId = Number.parseInt(selectedNode.id, 10);
     const path = selectedNode.path;
-    const selectedFolder: LfFolder = {
+    if (selectedNode.entryType === EntryType.Shortcut) {
+      if (selectedNode.targetId)
+      entryId = selectedNode.targetId;
+    }
+    let selectedFolder = {
```

Comment:
same thing as in the other PR, it's more robust to just save the path because that way if the entry is deleted, etc. we can handle that